src/Features/CSharp/Portable/UseUTF8StringLiteral/UseUTF8StringLiteralRefactoringProvider.cs

```diff
@@ -0,0 +1,192 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.CSharp.EmbeddedLanguages.VirtualChars;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Shared.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Operations;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.UseUTF8StringLiteral
+{
+    [ExportCodeRefactoringProvider(LanguageNames.CSharp, Name = PredefinedCodeRefactoringProviderNames.UseUTF8StringLiteral), Shared]
+    internal sealed class UseUTF8StringLiteralRefactoringProvider : CodeRefactoringProvider
+    {
+        private const char QuoteCharacter = '"';
+        private const string Suffix = "u8";
+
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public UseUTF8StringLiteralRefactoringProvider()
+        {
+        }
+
+        public override async Task ComputeRefactoringsAsync(CodeRefactoringContext context)
+        {
+            var (document, _, cancellationToken) = context;
+
+            var node = (SyntaxNode?)await context.TryGetRelevantNodeAsync<ArrayCreationExpressionSyntax>().ConfigureAwait(false) ??
+                await context.TryGetRelevantNodeAsync<ImplicitArrayCreationExpressionSyntax>().ConfigureAwait(false);
```

Comment:
This is new. Is it better to not use the helpers? Or use the helpers to get ExpressionSyntax and then manually compare to these two array expression types?